src/bootstrap/config.rs

```diff
@@ -987,10 +801,10 @@ impl Config {
             config.out = crate::util::absolute(&config.out);
         }
 
-        config.initial_rustc = build
-            .rustc
-            .map(PathBuf::from)
-            .unwrap_or_else(|| config.out.join(config.build.triple).join("stage0/bin/rustc"));
+        config.initial_rustc = build.rustc.map(PathBuf::from).unwrap_or_else(|| {
+            config.download_beta_toolchain();
```

Comment:
This feels like a significant change that shouldn't be coupled to factoring out a minimal configuration - do we need it in this PR? (Can we split this PR up?)

How did this used to work? I'm not seeing the previous call to this...

Reply:
It's also needed in this PR. I can split https://github.com/rust-lang/rust/pull/107812/commits/a2f975a4f079ea1f1f0eeab5038f630d2595eb1e this commit out to another PR. But we will need it merged first before taking this one on master.